aws/resource_aws_ses_identity_notification.go

```diff
@@ -0,0 +1,137 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ses"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsSesNotification() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsSesNotificationSet,
+		Read:   resourceAwsSesNotificationRead,
+		Update: resourceAwsSesNotificationSet,
+		Delete: resourceAwsSesNotificationDelete,
+
+		Schema: map[string]*schema.Schema{
+			"topic_arn": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateArn,
+			},
+
+			"notification_type": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateNotificationType,
+			},
+
+			"identity": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateIdentity,
+			},
+		},
+	}
+}
+
+func resourceAwsSesNotificationSet(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sesConn
+	topic := d.Get("topic_arn").(string)
+	notification := d.Get("notification_type").(string)
+	identity := d.Get("identity").(string)
+
+	setOpts := &ses.SetIdentityNotificationTopicInput{
+		Identity:         aws.String(identity),
+		NotificationType: aws.String(notification),
+		SnsTopic:         aws.String(topic),
+	}
+
+	log.Printf("[DEBUG] Setting SES Identity Notification: %#v", setOpts)
+
+	if _, err := conn.SetIdentityNotificationTopic(setOpts); err != nil {
+		return fmt.Errorf("Error setting SES Identity Notification: %s", err)
+	}
+
+	return resourceAwsSesNotificationRead(d, meta)
+}
+
+func resourceAwsSesNotificationRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sesConn
+	notification := d.Get("notification_type").(string)
```

Comment:
Ideally, we should be reading this information from the Terraform resource ID via `d.Id()`. If implemented as noted above, this can be accomplished via:

```go
identity, notificationType, err := decodeSesIdentityNotificationId(d.Id())
if err != nil {
  return err
}
```

And the function is implemented like:

```go
func decodeSesIdentityNotificationId(id string) (string, string, error) {
	parts := strings.Split(id, "|")
	if len(parts) != 2 {
		return "", "", fmt.Errorf("Unexpected format of ID (%q), expected IDENTITY|TYPE", id)
	}
	return parts[0], parts[1], nil
}
```